Gems/Archive/Code/Include/Archive/ArchiveInterfaceStructs.h

```diff
@@ -0,0 +1,274 @@
+        AZ::u64 m_tocOffset{};
+
+        //! Compressed Size of the Table of Contents
+        //! Max size is 512MiB or 2^29 bytes
+        //! The TOC offset + TOC offset is equal to total size of the Archive file
+        //! offset = 24
+        AZ::u32 m_tocCompressedSize : 29;
+        //! Compression algorithm used for the Table of Contents
+        //! The maximum 3-bit value of 7 is reserved for uncompressed
+        //! Other values count as a offset in the Compression Algorithm ID table
+        AZ::u32 m_tocCompressionAlgoIndex : 3;
+
+        //! Uncompressed size of the Table of Contents File Metadata section.
+        //! offset = 28
+        AZ::u32 m_tocFileMetadataTableUncompressedSize{};
+        //! Uncompressed size of the Table of Contents File Path index
+        //! The File Path index is used to lookup the location for a file path
+        //! within the archive
+        //! offset = 32
+        AZ::u32 m_tocPathIndexTableUncompressedSize{};
+        //! Uncompressed size of the Table of Contents File Path section
+        //! It contains a blob of FilePaths without any null-termination
+        //! The File Path Index entries are used to look up a file path
+        //! through using the path offset + size entry
+        //! offset = 36
+        AZ::u32 m_tocPathTableUncompressedSize{};
+        //! Uncompressed size of the Table of Contents File Block section
+        //! Contains compressed sizes of individual blocks of a file
+        //! In Archive V1 layout the block size is 2MiB
+        //! offset = 40
+        AZ::u32 m_tocBlockTableSize{};
+
+        //! Threshold value represents the cap on the size a block after it has been
+        //! sent through the compression step to determine if it should be stored compressed
+        //!
+        //! Due to block size defaulting to 2MiB, any blocks that are larger than 2_mib
+        //! after compression will be stored uncompressed.
+        //! So the maximum limit of this value is the Block Size
+        //! offset = 44
+        AZ::u32 m_compressionThreshold{ static_cast<AZ::u32>(ArchiveDefaultBlockSize) };
+
+        //! Stores 32-bit IDS of up to 7 compression algorithms that this archive can use
+        //! offset = 48
+        AZStd::array<Compression::CompressionAlgorithmId, 7> m_compressionAlgorithmsIds;
+
+        //! Offset from the beginning of the File Path index table
+        //! Where the first deleted block is located
+        //! offset = 76
+        AZ::u32 m_firstDeletedFileIndex{ DeletedPathIndexSentinel };
+
+        //! Offset from the beginning of the file block section to the first
+        //! deleted block.
+        //! The first 8 bytes of each deleted block will contain the offset to the next deleted block
+        //! or 0xffff'ffff'ffff'ffff if this is the last deleted block
+        //! offset = 80
+        AZ::u64 m_firstDeletedBlockOffset{ DeletedBlockOffsetSentinel };
+
+        //! total offset = 88
+
+        //! Max FileCount
```

Comment:
or maybe this is a better place for the warning about increasing max filecount

Reply:
Yeah, I can put your suggestion about the previous comment here.